samplers/checks.go

```diff
@@ -0,0 +1,67 @@
+package samplers
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/segmentio/fasthash/fnv1a"
+)
+
+// CheckStatus holds the statuses of all status checks that the veneur
+// has received, if a retention time for these checks is configured.
+type CheckStatus struct {
+	// statuses maps from one (UDPMetric-like) digest value for a
+	// status check to its success status.
+	statuses map[uint32]float64
+	resetAt  time.Time
+	validFor time.Duration
+}
+
+func NewCheckStatusTracker(validFor time.Duration) *CheckStatus {
+	return &CheckStatus{
+		statuses: make(map[uint32]float64),
+		resetAt:  time.Now().Add(validFor),
+		validFor: validFor,
+	}
+}
+
+func (c *CheckStatus) ShouldRecord(status *StatusCheck) bool {
```

Comment:
I'd like to see some comments on here. It was a bit jarring for me to figure out that calls to `ShouldRecord` are actually stateful, as the determination also changes the map's value (last, current).

Reply:
Ohhh yeah, forgot about the docs here! Thanks.